client/my-sites/plans-v2/products-grid-alt-2/index.tsx

```diff
@@ -0,0 +1,179 @@
+import { sortBy } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import isJetpackCloud from 'calypso/lib/jetpack/is-jetpack-cloud';
+import { JETPACK_LEGACY_PLANS } from 'calypso/lib/plans/constants';
+import { getCurrentUserCurrencyCode } from 'calypso/state/current-user/selectors';
+import getSitePlan from 'calypso/state/sites/selectors/get-site-plan';
+import getSelectedSiteId from 'calypso/state/ui/selectors/get-selected-site-id';
+import JetpackFreeCard from 'calypso/components/jetpack/card/jetpack-free-card-alt';
+import { slugToSelectorProduct } from '../utils';
+import useGetPlansGridProducts from '../use-get-plans-grid-products';
+import { isJetpackPlanSlug } from 'calypso/lib/products-values';
+import { getProductPosition } from '../product-grid/products-order';
+import ProductCardAlt2 from '../product-card-alt-2';
+import { getPlansToDisplay, getProductsToDisplay, isConnectionFlow } from '../product-grid/utils';
+
+/**
+ * Type dependencies
+ */
+import type { Duration, PurchaseCallback, QueryArgs } from '../types';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+const ProductsGridAlt2 = ( {
+	duration,
+	onSelectProduct,
+	urlQueryArgs,
+}: {
+	duration: Duration;
+	onSelectProduct: PurchaseCallback;
+	urlQueryArgs: QueryArgs;
+} ): ReactElement => {
+	const [ isPlanRowExpanded, setPlanRowExpanded ] = useState( true );
+	const [ isPlanRowWrapping, setPlanRowWrapping ] = useState( false );
+
+	const siteId = useSelector( getSelectedSiteId );
+	const currencyCode = useSelector( getCurrentUserCurrencyCode );
+	const currentPlanSlug =
+		useSelector( ( state ) => getSitePlan( state, siteId ) )?.product_slug || null;
+
+	const { availableProducts, purchasedProducts, includedInPlanProducts } = useGetPlansGridProducts(
+		siteId
+	);
+
+	const plansToDisplay = useMemo( () => getPlansToDisplay( { duration, currentPlanSlug } ), [
+		duration,
+		currentPlanSlug,
+	] );
+	const productsToDisplay = useMemo(
+		() =>
+			getProductsToDisplay( {
+				duration,
+				availableProducts,
+				purchasedProducts,
+				includedInPlanProducts,
+			} ),
+		[ duration, availableProducts, includedInPlanProducts, purchasedProducts ]
+	);
+	const sortedGridItems = useMemo(
+		() =>
+			sortBy( [ ...plansToDisplay, ...productsToDisplay ], ( item ) =>
+				getProductPosition( item.productSlug )
+			),
+		[ plansToDisplay, productsToDisplay ]
+	);
+	const planItems = useMemo(
+		() => sortedGridItems.filter( ( { productSlug } ) => isJetpackPlanSlug( productSlug ) ),
+		[ sortedGridItems ]
+	);
+	const productItems = useMemo(
+		() => sortedGridItems.filter( ( { productSlug } ) => ! isJetpackPlanSlug( productSlug ) ),
+		[ sortedGridItems ]
+	);
+
+	const planGridRef: RefObject< HTMLDivElement > = useRef( null );
+
+	const onPlanRowFeaturesToggle = useCallback( () => setPlanRowExpanded( ! isPlanRowExpanded ), [
+		isPlanRowExpanded,
+	] );
+	const onResize = useCallback( () => {
+		if ( planGridRef ) {
```

Comment:
Here, we fetch the number of items in the first row of the grid, and compare it to the number of plans.